ma-health/get-eligibility.js

```
@@ -0,0 +1,141 @@
+    3: [26328, 59377, 79164],
+    // TODO: add remaining:
+    4: [],
+    5: [],
+    6: [],
+    7: [],
+    8: []
+}
+
+/**
+ * Grabs the income level (1, 2, 3, or 4) for a given annual household income and household size
+ *
+ * @return {Number} Number will be 1, 2, 3 or 4
+ */
+function getIncomeLevel({ annualHouseholdIncome, householdSize }) {
+    if (!annualHouseholdIncome || !householdSize || householdSize > 8) {
+        return null;
+    }
+
+    const thresholds = INCOME_THRESHOLDS[householdSize];
+    let level;
+
+    for (let i = 0; i < thresholds.length; i++) {
+        if (annualHouseholdIncome < thresholds[i]) {
+            level = i + 1;
+            break;
+        }
+    }
+
+    return level;
+}
+
+/**
+ * An object detailing eligibility for various health programs
+ * E.g. {connectorCare: true, qualifiedHealthPlan: false, massHealth: false, massHealthCarePlus: false}
+ *
+ * @typedef EligibilityMatrix
+ * @property {Boolean} connectorCare
+ * @property {Boolean} qualifiedHealthPlan
+ * @property {Boolean} massHealth
+ * @property {Boolean} massHealthCarePlus
+ */
+
+/**
+ * Returns a recipient's eligibility matrix as determined by the given set of parameters
+ * TODO: I only developed this for the table pertaining to a household size of 3:
+ * https://www.bmchp.org/need-affordable-health-insurance/eligibility-guidelines/3
+ * We need to expand it to other household sizes, which might require restructuring the code a bunch.
+ *
+ * @return {EligibilityMatrix}
+ */
+function planType({
+    householdSize,
+    annualHouseholdIncome,
+    age,
+    usCitizen = false,
+    usAlien = false,
+    massResident = false,
+    pregant = false,
+    hiv = false,
+    disabled = false,
+    breastCancer = false,
+    cervicalCancer = false,
+    caretaker = false
+}) {
+    const incomeLevel = getIncomeLevel({ annualHouseholdIncome, householdSize });
+
+    if (!householdSize || !annualHouseholdIncome || !age || !incomeLevel) {
```

Is this just validating arguments? If you call a function without providing the necessary arguments, you should probably fail loudly so it's easier to catch errors.